pythonforandroid/bootstraps/common/build/build.py

```diff
@@ -297,9 +297,12 @@ def make_package(args):
     # Add extra environment variable file into tar-able directory:
     env_vars_tarpath = tempfile.mkdtemp(prefix="p4a-extra-env-")
     with open(os.path.join(env_vars_tarpath, "p4a_env_vars.txt"), "w") as f:
-        f.write("P4A_IS_WINDOWED=" + str(args.window) + "\n")
-        if hasattr(args, "orientation"):
-            f.write("P4A_ORIENTATION=" + str(args.orientation) + "\n")
+        try:
+            f.write("P4A_IS_WINDOWED=" + str(args.window) + "\n")
+            if hasattr(args, "orientation"):
+                f.write("P4A_ORIENTATION=" + str(args.orientation) + "\n")
+        except AttributeError as e:
+            print ('kivy is not in the requirements')
```

Comment:
Nice! Just some minor suggestions, we could simplify that code by killing the `hasattr()` check since we're catching `AttributeError` anyway. Also you can remove the `as e` local variable since you're not using it.
So I suggest the following then:
```python
        try:
            f.write("P4A_IS_WINDOWED=" + str(args.window) + "\n")
            f.write("P4A_ORIENTATION=" + str(args.orientation) + "\n")
        except AttributeError:
            print ('kivy is not in the requirements')
```

Reply:
..or I make an explicit check for args.window, perhaps thats cleaner

Reply:
Up to you, but I think it's more Pythonic to `EAFP`, see:
https://stackoverflow.com/a/11360880
In any case it's better to not mix both approaches indeed

Reply:
normally I also tend to EAFP, but here we have also to cover the case that we have 'window' and no 'orientatin', then the message would be misleading